### PR TITLE
TSK-942: Fix timing issue in CreateTaskAccTest

### PR DESCRIPTION
**Pull request by benjamineckstein**

### CODE REVIEW

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java [hunk 1]
```diff
@@ -176,19 +173,16 @@ public void testClaimTaskWithDefaultFlag()
...
         assertNotNull(claimedTask.getOwner());
         assertEquals(claimedTask.getOwner(), CurrentUserContext.getUserid());
         assertNotNull(claimedTask.getClaimed());
-        assertTrue(claimedTask.getClaimed().isAfter(before));
-        assertTrue(claimedTask.getClaimed().isBefore(after));
-        assertTrue(claimedTask.getModified().isAfter(before));
-        assertTrue(claimedTask.getModified().isBefore(after));
+        assertTrue(isBeforeOrEqual(before, claimedTask.getClaimed()));
+        assertTrue(isBeforeOrEqual(claimedTask.getCreated(), claimedTask.getClaimed()));
+        assertTrue(isBeforeOrEqual(claimedTask.getClaimed(), Instant.now()));
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java]: You are using before / claimedTask.getCreated() not consistant. Please use the before variable

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java [hunk 2]
```diff
@@ -211,14 +205,14 @@ public void testForceClaimTaskFromOtherUser()
...
 
         assertEquals(CurrentUserContext.getUserid(), taskAfterClaim.getOwner());
-        assertTrue(taskAfterClaim.getModified().isAfter(beforeForceClaim));
-        assertTrue(taskAfterClaim.getClaimed().isAfter(beforeForceClaim));
+        assertTrue(isBeforeOrEqual(beforeForceClaim, taskAfterClaim.getModified()));
+        assertTrue(isBeforeOrEqual(beforeForceClaim, taskAfterClaim.getClaimed()));
+        assertTrue(isBeforeOrEqual(taskAfterClaim.getCreated(), taskAfterClaim.getModified()));
+
         assertEquals(TaskState.CLAIMED, taskAfterClaim.getState());
-        assertNotEquals(createdTask.getCreated(), taskAfterClaim.getModified());
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java]: why did u remove this test?

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java [hunk 3]
```diff
@@ -211,14 +205,14 @@ public void testForceClaimTaskFromOtherUser()
...
         Instant beforeForceClaim = Instant.now();
-        waitAMillisecond();
         Task taskAfterClaim = taskService.forceClaim(createdTask.getId());
 
         assertEquals(CurrentUserContext.getUserid(), taskAfterClaim.getOwner());
-        assertTrue(taskAfterClaim.getModified().isAfter(beforeForceClaim));
-        assertTrue(taskAfterClaim.getClaimed().isAfter(beforeForceClaim));
+        assertTrue(isBeforeOrEqual(beforeForceClaim, taskAfterClaim.getModified()));
+        assertTrue(isBeforeOrEqual(beforeForceClaim, taskAfterClaim.getClaimed()));
+        assertTrue(isBeforeOrEqual(taskAfterClaim.getCreated(), taskAfterClaim.getModified()));
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java]: They have to be equal.

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java (hunk 2 above)
Comment (mustaphazorgati) [reply lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java]: Ignore this

File: lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java [hunk 4]
```diff
@@ -98,17 +98,23 @@ public void testCreateTaskWithPlanned()
...
         assertNotNull(createdTask.getPlanned());
-        assertEquals(createdTask.getCreated().plus(2, ChronoUnit.HOURS).truncatedTo(ChronoUnit.SECONDS),
-            createdTask.getPlanned().truncatedTo(ChronoUnit.SECONDS));
+        assertEquals(instantPlanned, createdTask.getPlanned());
+        assertTrue(createdTask.getCreated().isBefore(createdTask.getPlanned()));
+
+        //verify that planned takes place 2 hours after creation (+- 5 seconds)
+        long difference = Duration.between(createdTask.getCreated(), createdTask.getPlanned()).abs().getSeconds();
+        //add some tolerance to ignore that "created" depends on execution speed
+        assertTrue(Math.abs(difference - 2 * 60 * 60) < 5);
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java]: what are you doing here exactly? I expected difference to be in seconds already..

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java (hunk 3 above)
Comment (mustaphazorgati) [reply lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java]: yes.

File: lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java (hunk 4 above)
Comment (mustaphazorgati) [reply lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java]: for better readability: 
subtract 2 hours using https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#minusHours-long-

and then only check if < 5.

what do u think?

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java (hunk 3 above)
Comment (benjamineckstein) [reply lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java]: I don't get it. Why should created equals modify? We claimed the task after creation

File: lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java (hunk 4 above)
Comment (benjamineckstein) [reply lib/taskana-core/src/test/java/acceptance/task/CreateTaskAccTest.java]: done

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java (hunk 1 above)
Comment (mustaphazorgati) [reply lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java]: nevermind.

**Pull request merged**